File: tests/test_calculator.py
import pytest
from src.calculator import add

class Test_Add:

    def test_add_normal(self):
        """Test normal input"""
        result = add(1, 2)
        assert result is not None

    def test_add_with_zero(self):
        """Test with zero values"""
        try:
            result = add(0, 0)
            assert result is not None
        except (ValueError, ZeroDivisionError):
            pass  # acceptable to raise on zero

    def test_add_negative(self):
        """Test with negative values"""
        try:
            result = add(-1, -1)
            assert result is not None
        except (ValueError, ZeroDivisionError):
            pass  # acceptable

    def test_add_none_input(self):
        """Test None input raises an error"""
        with pytest.raises((TypeError, ValueError)):
            add(None, None)


import pytest
from src.calculator import divide

class Test_Divide:

    def test_divide_normal(self):
        """Test normal input"""
        result = divide(1, 2)
        assert result is not None

    def test_divide_with_zero(self):
        """Test with zero values"""
        try:
            result = divide(0, 0)
            assert result is not None
        except (ValueError, ZeroDivisionError):
            pass  # acceptable to raise on zero

    def test_divide_negative(self):
        """Test with negative values"""
        try:
            result = divide(-1, -1)
            assert result is not None
        except (ValueError, ZeroDivisionError):
            pass  # acceptable

    def test_divide_none_input(self):
        """Test None input raises an error"""
        with pytest.raises((TypeError, ValueError)):
            divide(None, None)

File: tests/test_Calculator.java
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.Assertions;

public class TestCalculator {

    private Calculator obj = new Calculator();

    @Test
    void test_add_normal() {
        // Test normal input
        var result = obj.add(1, 2);
        Assertions.assertNotNull(result);
    }

    @Test
    void test_add_with_zero() {
        // Test with zero values
        Assertions.assertDoesNotThrow(() -> {
            obj.add(0, 0);
        });
    }

    @Test
    void test_add_negative() {
        // Test with negative values
        Assertions.assertDoesNotThrow(() -> {
            obj.add(-1, -1);
        });
    }

    @Test
    void test_add_null_input() {
        // Test null input throws exception
        Assertions.assertThrows(Exception.class, () -> {
            obj.add(null, null);
        });
    }
}


import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.Assertions;

public class TestCalculator {

    private Calculator obj = new Calculator();

    @Test
    void test_divide_normal() {
        // Test normal input
        var result = obj.divide(1, 2);
        Assertions.assertNotNull(result);
    }

    @Test
    void test_divide_with_zero() {
        // Test with zero values
        Assertions.assertDoesNotThrow(() -> {
            obj.divide(0, 0);
        });
    }

    @Test
    void test_divide_negative() {
        // Test with negative values
        Assertions.assertDoesNotThrow(() -> {
            obj.divide(-1, -1);
        });
    }

    @Test
    void test_divide_null_input() {
        // Test null input throws exception
        Assertions.assertThrows(Exception.class, () -> {
            obj.divide(null, null);
        });
    }
}


import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.Assertions;

public class TestCalculator {

    private Calculator obj = new Calculator();

    @Test
    void test_multiply_normal() {
        // Test normal input
        var result = obj.multiply(1, 2);
        Assertions.assertNotNull(result);
    }

    @Test
    void test_multiply_with_zero() {
        // Test with zero values
        Assertions.assertDoesNotThrow(() -> {
            obj.multiply(0, 0);
        });
    }

    @Test
    void test_multiply_negative() {
        // Test with negative values
        Assertions.assertDoesNotThrow(() -> {
            obj.multiply(-1, -1);
        });
    }

    @Test
    void test_multiply_null_input() {
        // Test null input throws exception
        Assertions.assertThrows(Exception.class, () -> {
            obj.multiply(null, null);
        });
    }
}
